packages/typescript-estree/src/parseSettings/getProjectConfigFiles.ts

```diff
@@ -57,7 +57,7 @@ export function getProjectConfigFiles(
     checkedDirectories.push(directory);
   } while (
     directory.length > 1 &&
-    directory.length >= parseSettings.tsconfigRootDir.length
+    directory.length >= Math.max(1, parseSettings.tsconfigRootDir.replace(/[/\\]+$/, '').length)
```

Comment:
this expression will be re-evaluated every single loop which is wildly inefficient.
we should instead normalise the value when creating the `parseSettings` object so it's done exactly once.